zeppelin-test/src/main/java/org/apache/zeppelin/test/DownloadUtils.java

```diff
@@ -0,0 +1,537 @@
+  private static final String ARCHIVE_URL = "https://archive.apache.org/dist/";
+
+  private static String downloadFolder = System.getProperty("user.home") + "/.cache";
+  public static final String DEFAULT_SPARK_VERSION = "3.4.2";
+  public static final String DEFAULT_SPARK_HADOOP_VERSION = "3";
+
+  private DownloadUtils() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  static {
+    try {
+      FileUtils.forceMkdir(new File(downloadFolder));
+    } catch (IOException e) {
+      throw new RuntimeException("Fail to create download folder: " + downloadFolder, e);
+    }
+  }
+
+  /**
+   * Download Spark with default versions
+   *
+   * @return home of Spark installation
+   */
+  public static String downloadSpark() {
+    return downloadSpark(DEFAULT_SPARK_VERSION, DEFAULT_SPARK_HADOOP_VERSION);
+  }
+
+  /**
+   * Download of a Spark distribution
+   *
+   * @param sparkVersion
+   * @param hadoopVersion
+   * @return home of Spark installation
+   */
+  public static String downloadSpark(String sparkVersion, String hadoopVersion) {
+    File sparkFolder = new File(downloadFolder, "spark");
+    File targetSparkHomeFolder =
+        new File(sparkFolder, "spark-" + sparkVersion + "-bin-hadoop" + hadoopVersion);
+    return downloadSpark(sparkVersion, hadoopVersion, targetSparkHomeFolder);
+  }
+
+  /**
+   * Download of a Spark distribution
+   *
+   * @param sparkVersion
+   * @param hadoopVersion
+   * @param targetSparkHomeFolder - where should the spark archive be extracted
+   * @return home of Spark installation
+   */
+  public static String downloadSpark(String sparkVersion, String hadoopVersion,
+      File targetSparkHomeFolder) {
+    File sparkFolder = new File(downloadFolder, "spark");
+    sparkFolder.mkdir();
+    if (targetSparkHomeFolder.exists()) {
+      LOGGER.info("Skip to download Spark {}-{} as it is already downloaded.", sparkVersion,
+          hadoopVersion);
+      return targetSparkHomeFolder.getAbsolutePath();
+    }
+    File sparkTarGZ =
+        new File(sparkFolder, "spark-" + sparkVersion + "-bin-hadoop" + hadoopVersion + ".tgz");
+    try {
+      URL mirrorURL = new URL(
+          IOUtils.toString(new URL(MIRROR_URL), StandardCharsets.UTF_8) + generateDownloadURL(
+              "spark", sparkVersion, "-bin-hadoop" + hadoopVersion + ".tgz", "spark"));
+      URL archiveURL = new URL(ARCHIVE_URL + generateDownloadURL(
+              "spark", sparkVersion, "-bin-hadoop" + hadoopVersion + ".tgz", "spark"));
+      LOGGER.info("Download Spark {}-{}", sparkVersion, hadoopVersion);
+      download(new DownloadRequest(mirrorURL, archiveURL), sparkTarGZ);
+      ProgressBarBuilder pbb = new ProgressBarBuilder()
+          .setTaskName("Unarchiv")
+          .setUnit("MiB", 1048576) // setting the progress bar to use MiB as the unit
+          .setStyle(ProgressBarStyle.ASCII)
+          .setUpdateIntervalMillis(1000)
```

Comment:
we may want to use a large interval in CI console to avoid too many logs. suggest making it configurable via env var